Lib/inspect.py

```diff
@@ -2943,11 +2943,20 @@ def __init__(self, parameters=None, *, return_annotation=_empty,
                 params = OrderedDict()
                 top_kind = _POSITIONAL_ONLY
                 seen_default = False
+                seen_var_parameters = set()
 
                 for param in parameters:
                     kind = param.kind
                     name = param.name
 
+                    if kind in (_VAR_POSITIONAL, _VAR_KEYWORD):
+                        if kind in seen_var_parameters:
+                            msg = 'more than one {} parameter'
+                            msg = msg.format(kind.description)
+                            raise ValueError(msg)
```

Comment:
```suggestion
                    if kind in {_VAR_POSITIONAL, _VAR_KEYWORD}:
                        if kind in seen_var_parameters:
                            msg = f'more than one {kind.description} parameter'
                            raise ValueError(msg)
```

Reply:
I like f-string better too, but the format function is used next to the place where I make changes. Is it ok if this point is not very consistent?
Just below in this function:
```py
                    if kind < top_kind:
                        msg = (
                            'wrong parameter order: {} parameter before {} '
                            'parameter'
                        )
                        msg = msg.format(top_kind.description,
                                         kind.description)
                        raise ValueError(msg)
```

Reply:
Well.. I suggest we can change .format to f-strings or keep .format as you did. WDYT @serhiy-storchaka? do you prefer .format() or f-strings? (or inconsistent usage?) By the way, I don't remember whether .format() and f-strings are now almost equivalent in terms of performance but maybe it's the case (in which case .format() would be better).

Reply:
This code precedes f-strings. In new code we are free to use what looks better. I think that f-strings are more preferable in most cases (although there are exceptions).

Creating a set here is a preliminary optimization (which is actually a pessimization).

Reply:
Oh I thought that the set creation would be optimized by the interpreter and detected as a constant. Probably should have created the set outside of the loop but ok with keeping the tuple.